src/translate.js

```diff
@@ -26,12 +26,13 @@ export default (singleton) => function translate(text, plural, options) {
   finalOptions = {
     ...defaultOptions,
     ...finalOptions,
+    context: finalOptions && finalOptions.context ? finalOptions.context + '%04' : '',
   }
 
   const [
     translatedSingular,
     translatedPlural,
-  ] = (singleton.messages[text] || [null, null, null]).slice(1)
+  ] = (singleton.messages[unescape(finalOptions.context + text)] || [null, null, null]).slice(1)
```

Comment:
what's the `unescape` doing here? is it meant only for unescaping the `'%04'` separator char? If so, I would suggest to append it directly in an unescaped fashion at line 29, like:

```
finalOptions.context + '\u0004'
```

The current implementation will unescape the the whole key, which could be harmful.